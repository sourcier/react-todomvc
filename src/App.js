import React from 'react';
import { Route, Switch } from 'react-router-dom';
import {
  RecoilRoot,
} from 'recoil'
import 'todomvc-common/base.css'
import 'todomvc-app-css/index.css'

import Home from './Home';

const App = () => (
  <RecoilRoot>
    <Switch>
      <Route exact path="/" component={Home} />
    </Switch>
  </RecoilRoot>
);

export default App;
