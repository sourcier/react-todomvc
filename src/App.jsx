import React from "react";
import { Provider } from "react-redux";

import Home from "./Home";
import store from "./store";

const App = () => (
  <Provider store={store}>
    <Home />
  </Provider>
);

export default App;
