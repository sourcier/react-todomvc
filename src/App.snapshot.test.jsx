import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

describe("<App />", () => {
  test("renders without exploding", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
