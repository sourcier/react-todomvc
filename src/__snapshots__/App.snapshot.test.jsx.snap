// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without exploding 1`] = `
Array [
  <section
    className="todoapp"
  >
    <header
      className="header"
    >
      <h1>
        todos
      </h1>
      <form
        onSubmit={[Function]}
      >
        <input
          className="new-todo"
          onChange={[Function]}
          placeholder="What needs to be done?"
          value=""
        />
      </form>
    </header>
    <section
      className="main"
    >
      <ul
        className="todo-list"
      />
    </section>
  </section>,
  <footer
    className="info"
  >
    <p>
      Double-click to edit a todo
    </p>
    <p>
      Template by 
      <a
        href="http://sindresorhus.com"
      >
        Sindre Sorhus
      </a>
    </p>
    <p>
      Created by 
      <a
        href="http://sourcier.uk"
      >
        Roger Rajaratnam
      </a>
    </p>
    <p>
      Part of 
      <a
        href="http://todomvc.com"
      >
        TodoMVC
      </a>
    </p>
  </footer>,
]
`;
